src/groups/bmq/bmqp/bmqp_protocol.h

```diff
@@ -3598,15 +3598,8 @@ inline RdaInfo& RdaInfo::setPotentiallyPoisonous(bool flag)
 inline RdaInfo& RdaInfo::setCounter(unsigned int counter)
 {
     BSLS_ASSERT_SAFE(counter <= k_MAX_COUNTER_VALUE);
-    if (isUnlimited()) {
-        d_counter &= ~e_UNLIMITED;
-        d_counter += counter;
-    }
-    else {
-        d_counter &= ~k_MAX_COUNTER_VALUE;
-        d_counter += counter;
-    }
-
+    // Drop e_UNLIMITED bit flag if set, but save e_POISONOUS bit
+    d_counter = counter | (d_counter & e_POISONOUS);
```

Comment:
Is this functionally equivalent to the previous code?

Reply:
Yes, I checked it several times and I'm sure that it's equivalent.
However, I might add an extra unit test to remove any chances that it breaks something.

Reply:
UPD: we already have RdaInfo Breathing Test which tests different scenarios